lib/usbmux/transformer/usbmux-decoder.js

```diff
@@ -0,0 +1,32 @@
+import { parse } from '../../util/plist';
+import Stream from 'stream';
+
+const HEADER_LENGTH = 16;
+
+class UsbmuxDecoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+  }
+
+  _transform (data, encoding, callback) {
```

Comment:
can we make the method asynchronous to avoid having the callback argument?

Reply:
That's part of the Stream API, so it won't be possible.

https://nodejs.org/api/stream.html#stream_implementing_a_transform_stream